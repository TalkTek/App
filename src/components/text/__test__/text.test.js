import React from 'react'
import { shallow } from 'enzyme'
import Text from '../text'
import renderer from 'react-test-renderer'
import { H1, H2, H3, H4, H5 } from '../index'

const textProps = {
  fontSize: 12,
  bold: true,
  color: '#000'
}

const textStyle = [
  { fontSize: 12, fontWeight: '700', color: '#000' },
  undefined
]

describe('<Text />', () => {
  test('render test', () => {
    shallow(<H1 />)
    shallow(<H2 />)
    shallow(<H3 />)
    shallow(<H4 />)
    shallow(<H5 />)
  })

  it('should render Text Tag', () => {
    const wrapper = shallow(<Text />).dive()
    expect(wrapper.type()).toEqual('Text')
  })

  it('should have correct style', () => {
    const element = <Text {...textProps} />
    const wrapper = shallow(element).dive()
    const tree = renderer.create(element)
    expect(wrapper.prop('style')).toEqual(textStyle)
    expect(tree).toMatchSnapshot()
  })
  
  it('should have correct text color', () => {
    const element = <Text gray />
    const wrapper = shallow(element).dive()
    const tree = renderer.create(element)
    expect(wrapper.prop('style')[0].color).toEqual('rgb(158, 158, 158)')
    expect(tree).toMatchSnapshot()
  })

  test('render correct snapshot', () => {
    const tree = renderer.create(
      <Text />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})