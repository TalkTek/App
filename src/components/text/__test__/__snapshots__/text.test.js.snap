// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> render correct snapshot 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgb(33, 33, 33)",
        "fontSize": undefined,
        "fontWeight": null,
      },
      undefined,
    ]
  }
/>
`;

exports[`<Text /> should have correct style 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  bold={true}
  color="#000"
  ellipsizeMode="tail"
  fontSize={12}
  style={
    Array [
      Object {
        "color": "#000",
        "fontSize": 12,
        "fontWeight": "700",
      },
      undefined,
    ]
  }
/>
`;

exports[`<Text /> should have correct text color 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  gray={true}
  style={
    Array [
      Object {
        "color": "rgb(158, 158, 158)",
        "fontSize": undefined,
        "fontWeight": null,
      },
      undefined,
    ]
  }
/>
`;
